packages/protocol/contracts/shared/bridge/EtherBridgeWrapper.sol

```diff
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "./Bridge.sol";
+import "../../shared/based/ITaiko.sol";
+import "src/layer1/based/ITaikoInbox.sol";
+import "src/shared/libs/LibAddress.sol";
+import "src/shared/common/EssentialContract.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+contract EtherBridgeWrapper is EssentialContract {
+    using Address for address;
+    using LibAddress for address;
+
+    /// @dev Represents an operation to send Ether to another chain.
+    struct EtherBridgeOp {
+        // Destination chain ID.
+        uint64 destChainId;
+        // The owner of the bridge message on the destination chain.
+        address destOwner;
+        // Recipient address.
+        address to;
+        // Processing fee for the relayer.
+        uint64 fee;
+        // Gas limit for the operation.
+        uint32 gasLimit;
+        // Amount of Ether to be sent.
+        uint256 amount;
+        // Added solver fee
+        uint256 solverFee;
+    }
+
+    /// @dev Represents an operation to solve an Ether bridging intent
+    struct SolverOp {
```

Comment:
Is it possible to reuse the SolverOp defines in ERC20Vault (maybe extract to another file) and use `0x0` for `EtherBridgeOp` to indicate the asset is Ether.

Same recommendation for `EtherBridgeOp`.